tests/specs/test_jinja2_file.py

```diff
@@ -0,0 +1,20 @@
+import unittest
+
+from conda_env import env
+from conda_env.specs.jinja2_file import Jinja2Spec
+
+from .. import utils
+
+
+class TestJinja2(unittest.TestCase):
+    def test_no_environment_file(self):
+        spec = Jinja2Spec(name=None, filename='not-a-file')
+        self.assertEqual(spec.can_handle(), False)
+
+    def test_no_name(self):
+        spec = Jinja2Spec(filename=utils.support_file('jinja2.yml'))
+        self.assertEqual(spec.can_handle(), True)
+
+    def test_get_environment(self):
+        spec = Jinja2Spec(filename=utils.support_file('jinja2.yml'))
+        self.assertIsInstance(spec.environment, env.Environment)
```

Comment:
I think it would be interesting here to test which dependencies were loaded, and make `jinja2.yml` install different dependencies based on some variable that we may artificially put in `sys` or `os`.

Something like this for example:

```yml
dependencies:
  - flask
  {% if sys.conda_env_testing == '1' %}
  - jinja2
  {% else %}
  - yaml
  {% endif %}
```

```python
def test_get_environment(self):
    sys.conda_env_testing = '1'
    self.addCleanup(delattr, sys, 'conda_env_testing')
    spec = Jinja2Spec(filename=utils.support_file('jinja2.yml'))
    self.assertIsInstance(spec.environment, env.Environment)
    self.assertEqual(spec.dependencies['conda'], ['flask', 'jinja2'])

    sys.conda_env_testing = '2'
    spec = Jinja2Spec(filename=utils.support_file('jinja2.yml'))
    self.assertEqual(spec.dependencies['conda'], ['flask', 'yaml'])
```